odoo/api.py

```diff
@@ -130,6 +130,11 @@ def _check_description(self):
     return attrsetter('_constrains', args)
 
 
+def ondelete(*, at_uninstall):
+    """ Mark a method to be executed right before data deletion in BaseModel.unlink()."""
+    return attrsetter('_ondelete', at_uninstall)
```

Comment:
Might be a good idea to explain about the `at_uninstall` bit no?

Also how does it handle overriding e.g. A defines an @ondelete, B tries to override that method, is the ondelete annotation automatically inherited? Does it have to be explicitly re-specified? What happens if one flags one way and the override the other?

Reply:
Also on one hand the name is nice, on the other it doesn't match the name of the operation in Odoo.

Reply:
Well it uses the same mechanism as `depends`, `constrains` and such, so the decorator is not propagated (from what I can gather only `api.returns` is) and if you override method M from A in B, B.M will shadow A.M.

This means that you can both modify/relax and add new checks, although the intended usage would be for module B to define method N that adds a check on top of A.M instead of overriding A.M, I would not recommend modifying/relaxing checks from other modules.